File: game/src/main/java/me/firecloud/game/rule/DealCardNode.java
/**
 * 
 */
package me.firecloud.game.rule;

import java.util.List;
import java.util.Map;

/**
 * @author kkppccdd
 * @email kkppccdd@gmail.com
 * @date Oct 14, 2012
 * 
 */
public class DealCardNode extends ActionNode {

	/*
	 * (non-Javadoc)
	 * 
	 * @see
	 * me.firecloud.game.rule.ActionNode#perform(me.firecloud.game.rule.Event,
	 * me.firecloud.game.rule.GameContext)
	 */
	@Override
	protected boolean perform(Event event, GameContext gameContext)
			throws RuleException {
		List<Map<String, Object>> holdCards = (List<Map<String, Object>>) gameContext
				.get("desktop.holdCards");
		while (holdCards.size() > 5) {
			// deal each one a card

		}
	}

}

File: card/src/main/scala/me/firecloud/card/rule/DealCardNode.scala
/**
 *
 */
package me.firecloud.card.rule

/**
 * @author kkppccdd
 * @email kkppccdd@gmail.com
 * @date Oct 19, 2012
 *
 */
class DealCardNode(nextNode: Node) extends ActionNode(nextNode: Node) {

  override def perform(gameContext: GameContext, event: Event): Boolean = {

    while (gameContext.dealer.hand.size() > 5) {
      // deal all cards to each player until remaining 5 cards
      for (player <- gameContext.players) yield {
        player.hand.add(gameContext.dealer.hand.remove(gameContext.dealer.hand.size() - 1))
      }
    }
    true
  }

}